### ISSUE TITLE
Size of directories is not calculated

### ISSUE THREAD

**Issue by NightMachinery** [svenstaro/miniserve#306]
![image](https://user-images.githubusercontent.com/36224762/81748620-66a56080-94bf-11ea-9acc-51b32f0f64fa.png)

**Comment by wyhaya** [comment]
I don't think this function should be added, this requires recursive directory to get file size and then merge.

it will be very slow.

**Comment by svenstaro** [comment]
Eh let's reopen it. It could be fast enough and it could also be cached. It should be opt-in though for sure so that the host can't be DOSed by default.

**Comment by wyhaya** [comment]
@svenstaro If you encounter a directory like `node_modules`, there may be 10,000-100,000... files, is there any good way?

**Comment by svenstaro** [comment]
Yeah, how about setting a useful timeout per directory?

**Comment by wyhaya** [comment]
I think timeout is not good. If it cannot be calculated quickly, it will time out every time, which means that the server can never calculate the size of some folders. The timeout should be set on some unstable functions, such as network packets.

**Comment by NightMachinary** [comment]
The timeout can be cached as “-1”. Or “NA.” I personally don’t have that many files.

**Comment by zamu-flowerpot** [comment]
I came here to mention KDE's file count and glad it was already been mentioned.
